### ISSUE TITLE
Document `CPtrToPointer`

### ISSUE THREAD

**Issue by certik** [lfortran/lfortran#4178]
| CPtrToPointer(expr cptr, expr ptr, expr? shape, expr? lower_bounds)       

It converts a C pointer (CPtr ttype) in `cptr` to a Fortran array in `ptr`, one has to prescribe the `shape` (not sure what happens if it is not prescribed, probably `shape` should be required) and optionally `lower_bounds` (1 otherwise).

We should not have defaults in ASR, so `lower_bounds` should be made required.

The `ptr` is either DescriptorArray  or PointerToDataArray, we need to clarify that.

**Comment by gxyd** [comment]
I do see `CPtrToPointer` documented here: https://docs.lfortran.org/en/asr/asr_nodes/statement_nodes/cptrtopointer/ maybe missing answer for a few questions raised in the main issue.

Also, no mention of `lower_bounds` in `CPtrToPointer` node.

**Comment by certik** [comment]
@Pranavchiku does `CPtrToPointer` always return the pointer with a physical type DescriptorArray?

**Comment by Pranavchiku** [comment]
Yes,this can be found from line `llvm::Value* fptr_des = arr_descr->get_pointer_to_dimension_descriptor_array(llvm_fptr);` in `visit_CPtrToPointer` of `asr_to_llvm.cpp`
